### PR TITLE
Fix `create` for a bare orphan in a section not adding random hex

### PR DESCRIPTION
**Pull request by SmileyChris**
# Description

When running `towncrier create section/+.feature`, the random hex isn't appended.

# Checklist

* [x] Make sure changes are covered by existing or new tests.
* [x] For at least one Python version, make sure local test run is green.
* [x] Create a file in `src/towncrier/newsfragments/`. Describe your
  change and include important information. Your change will be included in the public release notes.
* [x] Make sure all GitHub Actions checks are green (they are automatically checking all of the above).
* [x] Ensure `docs/tutorial.rst` is still up-to-date.

### CODE REVIEW

File: src/towncrier/test/test_create.py [hunk 1]
```diff
@@ -172,17 +172,39 @@ def test_create_orphan_fragment(self):
         """
         runner = CliRunner()
 
+        fragments = []
         with runner.isolated_filesystem():
```
Comment (hynek) [inline src/towncrier/test/test_create.py]: Could this be `@with_isolated_runner` instead?

File: src/towncrier/test/test_create.py [hunk 2]
```diff
@@ -172,17 +172,39 @@ def test_create_orphan_fragment(self):
...
+            self.assertEqual(0, result.exit_code)
+            result = runner.invoke(
+                _main, ["subsection/+.feature"], catch_exceptions=False
+            )
+            self.assertEqual(0, result.exit_code, result.output)
+
+            for _, dirs, files in os.walk("foo/newsfragments", topdown=False):
+                for file in files:
+                    fragments.append("/".join(dirs + [file]))
+                    print(dirs)
```
Comment (hynek) [inline src/towncrier/test/test_create.py]: oops

File: src/towncrier/test/test_create.py [hunk 3]
```diff
@@ -172,17 +172,39 @@ def test_create_orphan_fragment(self):
...
+            result = runner.invoke(_main, ["+.feature"])
+            self.assertEqual(0, result.exit_code)
+            result = runner.invoke(
+                _main, ["subsection/+.feature"], catch_exceptions=False
+            )
+            self.assertEqual(0, result.exit_code, result.output)
+
+            for _, dirs, files in os.walk("foo/newsfragments", topdown=False):
+                for file in files:
+                    fragments.append("/".join(dirs + [file]))
```
Comment (hynek) [inline src/towncrier/test/test_create.py]: Given you're using os.path everywhere, this should probably be something like `os.path.join(*dirs, file)`?

**Review by hynek — CHANGES_REQUESTED** [review]
Since you're now handling prefixes of different lengths, do we have a test for that case? It's easy to catch an off-by-1 here.

File: src/towncrier/test/test_create.py (hunk 1 above)
Comment (SmileyChris) [reply src/towncrier/test/test_create.py]: Sure, I was just copying the style of the tests surrounding when I wrote this originally, but I'll switch to that decorator.

File: src/towncrier/test/test_create.py (hunk 2 above)
Comment (SmileyChris) [reply src/towncrier/test/test_create.py]: Too used to pytest that just hides print statements unless the test fails :P

File: src/towncrier/test/test_create.py (hunk 3 above)
Comment (SmileyChris) [reply src/towncrier/test/test_create.py]: Again, the style of the other surrounding tests hard code '/', and I was trying to minimize changes - but sure!

**Comment by SmileyChris** [comment]
Test updated, and new test added to check for custom orphan prefix (of different length)

**Review by hynek — APPROVED** [review]
thanks!

**Pull request merged**